### ISSUE TITLE
All entries erased from the contact journal

### ISSUE THREAD

**Issue by achisto** [corona-warn-app/cwa-app-ios#3896]
## Technical details

- Device name: iPhone Xs
- iOS version: 15.1
- App version: 2.14.1 (0)

## Describe the bug

All entries from the contact journal are gone. 
Everything that's missing:
- journal entries
- persons
- locations
- rapid tests in the journal

So basically it looks like there was never anything entered into the journal ever. I discovered the problem today at 2021-11-26 19:34 CET. The last time I for sure known that the entries where still there is yesterday, 2021-11-25 at around 23:00 (+/- 2 hours). 

I did not update CWA since then. The last update of CWA is from three days ago, 2021-11-23. I did not delete the CWA, my certificates are still there (even one test certificate from 2021-11-19 that is still displayed in the certificate tab but no longer in the contact journal).

I had about 5-10 entries in the diary for the last 14 days, about 80 people in the persons list and about 50 locations in the locations list.

There is a file in my device's settings called "ANECompilerService.cpu_resource-2021-11-26-023358.ips" that I am happy to provide via e-mail.

## Steps to reproduce the issue

Here are some screenshots of the empty journal and the certificates section.
[contact journal 01](https://user-images.githubusercontent.com/20704439/143622447-a87453f4-d560-4e49-9638-787871175e72.PNG)
[contact journal 02](https://user-images.githubusercontent.com/20704439/143622451-e8030445-8a09-4d8a-a5f1-09b085cc232b.PNG)
[contact journal 03](https://user-images.githubusercontent.com/20704439/143622458-23cd9879-6273-4da3-929c-2e115013dc7b.PNG)
[persons](https://user-images.githubusercontent.com/20704439/143622459-9863c227-89cf-412d-84c4-a340845754ce.PNG)
[locations](https://user-images.githubusercontent.com/20704439/143622461-e66ffaa5-e0c9-4290-9bf5-a3f42d07e41d.PNG)
[certificates](https://user-images.githubusercontent.com/20704439/143622464-aabc62e9-ed26-4a50-95d6-57b81e9cea1d.PNG)

I don't think it's possible but I would really appreciate if there was a way to recover my entries. The CWA's contact diary is the only place I keep track of the people I meet and the places I visit so if I should get a red warning in the nearer future or needed to inform my contacts because of an infection on my side there was no way to really know for sure, I'd actually remembered everyone. A behavior like this should never happen under any circumstances and I am severely upset that I not only had to write this lengthy bug report but also can't trust CWA anymore and therefore need to use my notes app again. Very disappointing.

## Expected behaviour

Open the contact journal, see my entries, add new ones from the list of people/locations I've already used before.

## Possible Fix

No idea. 

As mentioned above, there is an incident report in my device's settings. Because I'm not sure if it contains private information, I would rather send it via e-mail to the CWA maintainers than publish it to Github,

Not sure if related but there is another issue with a similar problem: #3317 

Because of that one I checked my device's free space which seems okay to me: [screenshot free space](https://user-images.githubusercontent.com/20704439/143622715-81465526-4cb1-4e62-a2d7-7736619e4fc9.jpg)

**Comment by dsarkar** [comment]
Internal Tracking ID: [EXPOSUREAPP-10759](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10759)

**Comment by dsarkar** [comment]
Good morning @achisto, thanks for the very detailed report. We pinged a dev. He might come back to you regarding the error report. Best wishes, DS 

---
Corona-Warn-App Open Source Team

**Comment by jajoho** [comment]
FYI: I've had the same issue, but was unable to produce this 

#3317

**Comment by dsarkar** [comment]
Hi @jajoho,

Thanks for your comment. You had this issue while updating from version 2.15 to version 2.16?

Best wishes, DS

---
Corona-Warn-App Open Source Team

**Comment by jajoho** [comment]
Hi @dsarkar,

no. I had the issue with version 2.6.3 (in August 2021). But from the description it seems pretty similar to what happened to me at that time. The deletion was not related to any update.

Also with issue #3680 someone reported a similar behaviour.

In both cases, a possible cause of the problem was insufficient free memory.

@achisto Do you remember how much free space you had back then?

**Comment by achisto** [comment]
> 
> @achisto Do you remember how much free space you had back then?
> 
> 

Yes, the log file reads `Free disk space:  2629.99 MB/238.31 GB`

So around 2.5 GB still left.

**Comment by dsarkar** [comment]
@jajoho  Thanks for the feedback.
@jajoho @achisto Since then you did not have this issue again, i.e. it occurred once?

**Comment by achisto** [comment]
> 
> @jajoho @achisto Since then you did not have this issue again, i.e. it occurred once?

Correct, this was the first and only time it happened. In the aftermath of the event I also did free up some space on my iPhone as a precaution so that one possible cause for it to happen again is reduced as much as possible.

**Comment by dsarkar** [comment]
FYI PR: https://github.com/corona-warn-app/cwa-app-ios/pull/4389

**Comment by Ein-Tim** [comment]
@dsarkar Fell free to also add the https://github.com/corona-warn-app/cwa-app-ios/labels/Fix%202.21 label to the issue I just closed (https://github.com/corona-warn-app/cwa-app-ios/issues/3680).
